index.js

```diff
@@ -157,7 +158,7 @@ function compression(options) {
 
       // compression method
       var accept = accepts(req)
-      var method = accept.encoding(['gzip', 'deflate', 'identity'])
+      var method = accept.encoding(available)
 
       // negotiation failed
       if (!method || method === 'identity') {
```

Comment:
right
if you want disable `deflate` algorithm, you should use `['gzip', 'identity']`